packages/blade/src/components/Badge/Badge.tsx

```diff
@@ -0,0 +1,107 @@
+import type { ReactElement } from 'react';
+import type { CSSObject } from 'styled-components';
+import styled from 'styled-components';
+import { StyledBadge } from './StyledBadge';
+import { useTheme } from '~components/BladeProvider';
+import type { IconComponent } from '~components/Icons';
+import { BaseText } from '~components/Typography/BaseText';
+import { getPlatformType } from '~utils';
+import Box from '~components/Box';
+
+type BadgeProps = {
+  /**
+   * Sets the label for the badge.
+   *
+   */
+  children: string;
+  /**
+   * Sets the variant of the badge.
+   *
+   * @default 'neutral'
+   */
+  variant?: 'positive' | 'negative' | 'notice' | 'information' | 'neutral';
+  /**
+   * Sets the contrast of the badge.
+   *
+   * @default 'low'
+   */
+  contrast?: 'low' | 'high';
+  /**
+   * Sets the size of the badge.
+   *
+   * @default 'medium'
+   */
+  size?: 'small' | 'medium';
+  /**
+   * Icon to be displayed in the badge.
+   * Accepts a component of type `IconComponent` from Blade.
+   *
+   */
+  icon?: IconComponent;
+  /**
+   * Sets the fontWeight of the label.
+   *
+   * @default 'regular'
+   */
+  fontWeight?: 'regular' | 'bold';
+};
+
+const StyledBaseText = styled(BaseText)(
+  (): CSSObject => {
+    if (getPlatformType() !== 'react-native') {
```

Comment:
Optional: Can we move platform specific styles / logic to `.web` module? In favor of keeping conditional logic cleaner

Reply:
Wait I never noticed this. But we are only importing styled from `styled-components` right? 
So when this code runs on native doesn't it throw some kind of warning that we are not importing from `styled-components/native`? 

@chaitanyadeorukhkar

Reply:
I thought it automatically handled that 😹

Reply:
Haha for react native, we alias `styled-components` to `styled-components/native` 😅  

So for such small cases its okay. I prefer doing `styled-components/native` when we're styling a native specific thing like View or Pressable. But for a component like BaseText its okay since we know BaseText internally will handle platform specific things